### ISSUE TITLE
SOAP request structure does not conform to the specification

### ISSUE THREAD

**Issue by fazed** [tig-nl/buckaroo-magento2#44]
## Please follow the guide below

- Put an `x` into all the boxes [ ] relevant to your *issue* (like this: `[x]`)
- Use the *Preview* tab to see what your issue will actually look like.
- We may ask some questions or ask you to provide addition information after you placed your request.

---

### Make sure you are using the *latest* version: https://tig.nl/buckaroo-magento-extensies/
Issues with outdated version will be rejected.
- [x] I've **verified** and **I assure** that I'm running the latest version of the TIG Buckaroo Magento 2 extension.

---

### What is the purpose of your *issue*?
- [ ] Feature request (request for a new functionality)
- [x] Bug report (encountered problems with the TIG Buckaroo Magento 2 extension)
- [ ] Extension support request (request for adding support for a new extension)
- [ ] Other

---

### If the purpose of your issue is a *bug report*
We received multiple errors in our exception log regarding failed XML structure validation, see the error below for more information.

`[2018-11-27 10:21:13] main.CRITICAL: Structure validation of XML failed. The SOAP request structure does not conform to the specification. {"exception":"[object] (SoapFault(code: 0): Structure validation of XML failed. The SOAP request structure does not conform to the specification. at /xxx/vendor/tig/buckaroo/Soap/Client/SoapClientWSSEC.php:33)"} []`

We only occasionally get this error which causes us to suspect customer data is not correctly formulated/escaped when passing it to Buckaroo. In turn this casuses (payment) requests not being send, which is IMO a fairly critical issue.

Unfortunately, I can't provide more info since the exception message doesn't specify which order/payment/data causes the error. I've enabled debug logging on our affected environment and I'll update this issue with a debug log entry when the error occurs again.

Steps to reproduce:
Unknown for now, waiting for a related debug log entry.

If you need further information, please let me know.

### TIG support

- Please ask your question in English to ensure that your issue can help other people internationally. Nevertheless we will respond in English.

**Comment by tig-jaspersmits** [comment]
Hi @fazed,

Unfortunate to hear that you are experiencing this problem. I also understand that this is quite critical. But I need to be able to reproduce this since it is an unknown issue at this time. If you include the logging and exact steps to reproduce I will test this on a vanilla Magento environment.

Looking forward to your response.

Kind regards,

Jasper
TIG

**Comment by tig-jaspersmits** [comment]
Hi @fazed,

We didn't receive a response therefore I will close this issue. If there is still a problem please let us know!

Kind regards,

Jasper
TIG

**Comment by fazed** [comment]
Hi @tig-jaspersmits,

We have managed to get a hold of the bodies being send/received by the plugin. It's been awhile, finding a way to successfully log this data took some extra effort.

Sadly the data contains data which I would rather not publish publicly, but the data might prove to be the culprit. From what I've gathered it looks like fields containing non-alphanumeric characters (with some exceptions) are causing problems.

Moreover we've found that orders struck by this issue do _not_ appear in Magento (nor Buckaroo). People might not even notice they are experiencing this issue until they check with the exception log or having a customer complain about their order nog completing.

Currently we have an agent assigned to this case over at Buckaroo, which will get us in contact with a TIG employee to further investigate the problem. This is just a quick update for people who are experiencing the same issue(s).
